corehq/apps/locations/util.py

```diff
@@ -272,6 +272,9 @@ def purge_locations(domain):
             case.delete()
         loc.delete()
 
-    domain_obj = Domain.get_by_name(domain)
-    domain_obj.location_types = []
-    domain_obj.save()
+    db = Domain.get_db()
+    domain_obj = Domain.get_by_name(domain)  # cached lookup is fast but stale
+    domain_json = db.get(domain_obj._id)  # get latest raw, unwrapped doc
```

Comment:
Not crazy about the double lookup, but seemed like a good level of ease-of-implementation/understanding to get the latest version of the doc, unwrapped.  Happy to hear other suggestions if you've got 'em.